### DISCUSSION TITLE
Proposal for a Kotlin DSL to declare string resources instead of XML

### DISCUSSION THREAD

**Discussion by xxfast** [icerockdev/moko-resources#334] [Ideas]
Is there a reason why XML was chosen as the format to declare common string sets? 

Before I started using moko-resource, I had this in-house localization system to localize strings. I soon gave this up in favor of `moko-resources`. However, I quite liked the format I came up with
```kotlin
// :core/commonMain/kotlin
data class Localizable<T>(
  val default: T,
  val en: T = default,
  val en_AU: T = en,
  val si: T = default,
  // rest of the supported language sets
)

// :login/commonMain/resources
val strings: Localizable<LoginStrings> = Localizable(
  default = LoginStrings(
    ..
    invalidCredentialsSnackbar = "Invalid username or password",
    failedMessageSnackbar = { action, reason -> "Failed to $action due to a $reason" },
    ..
  ),
  si = LoginStrings(
    invalidCredentialsSnackbar = "වලංගු නොවන පරිශීලක නාමය හෝ මුරපදය",
    failedMessageSnackbar = { action, reason -> "$reason නිසා $action වෙත අසමත් විය" },
    ..
  ),
  // declarations for the rest of the language sets
)
```

 It would be awesome if moko-resource can parse this definition and generate platform-specific language files like 
 ```xml
  <string name="invalidCredentialsSnackbar">Invalid username or password</string>
  <string name="failedMessageSnackbar">Failed to %1$s due to a %2$s</string>
 ```
 
One obvious advantage over XML here is the type safety for arguments. Apart from that it is really just kotlin suger

**Comment by Alex009** [comment]
xml was used because it can be reused from android and it simple to read.
goal of moko-resources is generate for each platform true native localizations, that will be processed by target as original native localizations
